src/main/java/com/example/bookmyshow/service/impl/UserContextServiceImpl.java

```diff
@@ -0,0 +1,29 @@
+package com.example.bookmyshow.service.impl;
+
+import com.example.bookmyshow.model.User;
+import com.example.bookmyshow.service.UserContextService;
+import org.springframework.context.annotation.Scope;
+import org.springframework.context.annotation.ScopedProxyMode;
+import org.springframework.stereotype.Service;
+import org.springframework.web.context.WebApplicationContext;
+
+@Service
+@Scope(value = WebApplicationContext.SCOPE_REQUEST,
```

Comment:
What is SCOPE_REQUEST. Why we are using it.
What is the use of @Scope annotations. 
How many types of bean scopes present

Reply:
scope request means it is a request specific terminology.In each request spring will create a new bean.We are using it because to store request specific data and handling request.  @Scope  define the scope of a bean.There are 6 types of scope. lets take singletone .for this only one bean will created and shared across the spring context.